src/analysis/calculus/deriv.lean

```diff
@@ -430,6 +443,53 @@ lemma deriv_within_of_open (hs : is_open s) (hx : x ∈ s) :
   deriv_within f s x = deriv f x :=
 by { unfold deriv_within, rw fderiv_within_of_open hs hx, refl }
 
+lemma deriv_mem_iff {f : 𝕜 → F} {s : set F} {x : 𝕜} :
+  deriv f x ∈ s ↔ (differentiable_at 𝕜 f x ∧ deriv f x ∈ s) ∨
+    (0 : F) ∈ s ∧ ¬differentiable_at 𝕜 f x :=
```

Comment:
The order seems weird, I would swap one of the `and`s in the RHS, probably the second one (same for other lemmas)